kiosk/src/Components/ScanQR.tsx

```diff
@@ -2,36 +2,96 @@ import { useState } from "react";
 import { Kiosk } from "../Models/Kiosk";
 import "../Kiosk.css";
 import play from "./QrScanner";
+import { addGameToKioskAsync } from "../BackendRequests";
+import { KioskState } from "../Models/KioskState";
+
 interface IProps {
     kiosk: Kiosk
 }
 
 const ScanQR: React.FC<IProps> = ({ kiosk }) => {
     const fullUrlHash = window.location.hash;
     const urlHashList = /add-game:((?:[a-zA-Z0-9]{6}))/.exec(fullUrlHash);
+    const screenWidth = window.screen.width;
+    console.log(screenWidth);
+    const phoneWidth = screenWidth < 500;
+    console.log(phoneWidth);
     const kioskId = urlHashList?.[1];
     const [scannerVisible, setScanner] = useState(false);
+    const [linkError, setLinkError] = useState(false);
+    const perisistentShareLen = 52;
+    const regularShareLen = 34;
 
     const renderQrScanner = () => {
         setScanner(true);
         play(kiosk, kioskId!);
     }
 
+    const checkUrl = async () => {
+        const input = document.getElementById("kiosk-share-link") as HTMLInputElement;
+        const inputValue = input.value;
+        const shareLink = /(https:\/\/)((arcade\.makecode\.com\/)|(makecode\.com\/))((?:S\d{5}-\d{5}-\d{5}-\d{5})|(?:_[a-zA-Z0-9]+))/.exec(inputValue);
+        let shareId;
+        if (shareLink) {
+            setLinkError(false);
+            shareId = /\/([^\/]+)\/?$/.exec(inputValue)?.[1];
+            try {
+                await addGameToKioskAsync(kioskId, shareId);
+                kiosk.navigate(KioskState.QrSuccess);
+            } catch (error) {
+                console.log("Unable to add game to kiosk. Please try again later");
+            }
+        } else {
+            setLinkError(true);
+        }
+
+    }
+
     return (
         <div className="scanQrPage">
             <h2>Add game to </h2>
-            <h2>Kiosk {kioskId}</h2>
+            <h1>Kiosk {kioskId}</h1>
             <div className="scanInstructions">
                 <ol>
-                    <li>Open your game on your computer</li>
+                    <li>Open your game on the arcade editor</li>
```

Comment:
Instead of being verbose here. Can we just put a link to "How do I get the share code of game?" link which opens a documentation page similar to multiplayer? 
![image](https://user-images.githubusercontent.com/6107272/202332295-fce84734-e521-4cde-808b-b8fb97dacf05.png)

All these documentation work great on phone too. 
https://arcade.makecode.com/multiplayer#host-game

Reply:
This page can be as simple as enter the share code or scan the QR code of the game.